src/main/java/util/FileWatcher.java

```diff
@@ -0,0 +1,100 @@
+package util;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.*;
+
+/**
+ * The Class FileWatcher.
+ * <p>
+ * A utility class for watching files for changes.
+ *
+ * @author Juhan Oskar Hennoste
+ * @since 0.0.3
+ */
+
+public class FileWatcher implements AutoCloseable {
+
+    private final WatchService watchService;
+    private final Path relativeTargetFile;
+
+    private final Logger log = LogManager.getLogger(FileWatcher.class);
+
+
+    public FileWatcher(Path targetFile) {
+        Path normalizedTargetFile = targetFile.toAbsolutePath().normalize();
+        Path targetFileFolder = normalizedTargetFile.getParent();
+
+        try {
+            this.watchService = FileSystems.getDefault().newWatchService();
+            targetFileFolder.register(watchService, StandardWatchEventKinds.ENTRY_MODIFY);
+            this.relativeTargetFile = targetFileFolder.relativize(normalizedTargetFile);
+        } catch (IOException e) {
+            // Reasonably certain this should not happen unless your setup is very broken
+            throw new UncheckedIOException("Failed to create file watcher for " + normalizedTargetFile, e);
+        }
+    }
+
+
+    /**
+     * Waits until the target file is modified (creating or deleting the file also counts as a modification)
+     */
+    public void waitForModified() throws InterruptedException {
+        // Wait for event corresponding to the desired file
+        boolean modifiedEventFound = false;
+        while (!modifiedEventFound) {
+            WatchKey key = watchService.take();
+            for (WatchEvent<?> event : key.pollEvents()) {
+                if (event.context().equals(relativeTargetFile)) {
+                    modifiedEventFound = true;
+                }
+            }
+            key.reset();
+        }
```

Comment:
Why implement all this from scratch and not use a library for it? I think the library that was removed would actually be pretty neat for file-watching because it has all these methods built in already.

Reply:
The implementation was copied from the existing implementation in GobPieConfReader and adjusted to work with repeated calls, so I already had the implementation on hand, I did not implement it from scratch.

I ultimately chose the somewhat low-level WatchService over the other two file watchers used in the project because using it made it easy to create a simple API with a wait and check function that was well suited to the current usage. If I recall correctly the methvin/directory-watcher API did not have a readily available check method equivalent and the Apache FileAlterationObserver api did not have a readily available wait method equivalent.

Also, the removed library (methvin/directory-watcher) was, as far as I can tell, for recursively watching directories, not for watching files.